src/katgpucbf/xbgpu/kernels/beamform.mako

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright (c) 2023, National Research Foundation (SARAO)
+ *
+ * Licensed under the BSD 3-Clause License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ *   https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+<%include file="/port.mako"/>
+
+#define N_BEAMS ${len(beam_pols)}
+#define BLOCK_TIME ${block_time}
+#define BLOCK_CHANNELS ${block_channels}
+#define QMAX 127
+
+// Quantise, and return saturation flag
+// TODO: this is copied from postproc.py. Create a shared implementation.
```

Comment:
Almost forgot to ask, if you haven't already, please create a ticket for this.

Reply:
Good idea. I've created NGC-1162.